### ISSUE TITLE
$ANDROID_SERIAL env variable is ignored

### ISSUE THREAD

**Issue by nnez140814** [Genymobile/scrcpy#3111]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: ArchLinux
 - scrcpy version: 1.23-1
 - installation method: pacman
 - device model: N/A
 - Android version: N/A

**Describe the bug**

The previous version(1.22-2) allows me to specify which device I want to cast by setting `$ANDROID_SERIAL` environment variable. In version 1.23-1, `$ANDROID_SERIAL` doesn't work anymore. If more than one device is connected, scrcpy will print this error and exit.

```
scrcpy 1.23 <https://github.com/Genymobile/scrcpy>
ERROR: Multiple (2) ADB devices:
ERROR:     -->   (usb)  xxxxxxx                        device  xxxxxx
ERROR:     --> (tcpip)  yy:5555                        device  yyyyyy
ERROR: Select a device via -s (--serial), -d (--select-usb) or -e (--select-tcpip)
ERROR: Server connection failed
```

**Comment by rom1v** [comment]
The `ANDROID_SERIAL` environment variable was never read by scrcpy (only by adb), so it worked only as a side effect when scrcpy called `adb` without explicit serial.

Now (since #3005), scrcpy handles the device selection on its own, and always calls `adb` with an explicit `-s` parameter. That's why the side effect does not work anymore.

We could explicitly read this environment variable if no selector is specified: #3113

Please review/test.

**Comment by rom1v** [comment]
Fixed by #3113.
